### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by rasberik**
## Pull Request Type

- [ ] Bug fix
- [ ] New feature
- [ ] Refactoring
- [x] Documentation update
- [ ] Chore

## Description
Update documentation and syntactic & grammar structure of mainly Context and PropertyWrapper parts, and files related to the these changes. 

### Some examples of repetitive pattern alignments in this PR:
- "_noun_ that to _verb_" - aligned as "_noun_ to _verb_"
    - Example: "operator that to reset an asynchronous atom value"  → "operator to reset an asynchronous atom value"
- "watch to" phrasal verb seems little off, suggesting to use regular "watch" verb
    - Example: "no longer watched to from anywhere" → "no longer watched from anywhere"
- Preserving original verb form:
    - Example: 
        - "to read, write, and otherwise interacting" → "to read, write, and otherwise interact"
        - "Refreshes and then return" → "Refreshes and then returns"
- Method Parameters changes:
     - "The default timeout interval is `nil`" → "The default timeout interval is `nil` which indicates no timeout."
     - Missing colon in " - Parameters:"
- "Parameter atom: An atom that associates the value." (repetitive) → find it confusing, since atoms do associate with value, but not always other way around. 
     - context specific definition added
- "notifies update to downstream atoms" (repetitive) → removed _update_ or rephrased according to situation
- `atoms or/and views` - using _or_ & _and_ have subtle difference, so I reviewed all cases and updated respectfully.
    - Example 1: 
        - ❌ "notifies downstream atoms or views" - can be understood as it updates only one of them, either atoms or views depending on situation.
        - ✅ "notifies downstream atoms and views": clear that it notifies everything downstream, whether its atom or view.
    - Example 2: 
        - ✅ "cancelled by downstream atoms or views", if any atom or view cancelled
        - ❌ "cancelled by downstream atoms and views" all of the downstream atoms and views cancelled
- Some sentences are rephrased.

### Remaining potential improvement points:

- Remove pronouns, e.g. "you" → In documentation, might be better avoid using pronouns, e.g.
    - "When you assign a new value" → "When it (atom) is assigned a new value"
    - "A modifier that you apply to an atom" → "A modifier that is applied to an atom"

May be can address them in separate PR

## Motivation and Context
To keep documentation up to date and easier to read.

## Impact on Existing Code
None

### CODE REVIEW

File: Sources/Atoms/Context/AtomTransactionContext.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
-/// A context structure that to read, watch, and otherwise interacting with atoms.
+/// A context structure to read, watch, and otherwise interact with atoms.
 ///
-/// Through this context, watching of an atom is initiated, and when that atom is updated,
-/// the value of the atom to which this context is provided will be updated transitively.
+/// When an atom watching is initiated through this context, and that atom is updated,
```
Comment (rasberik) [inline Sources/Atoms/Context/AtomTransactionContext.swift:3]: [Options]
1. When an atom watching is initiated through this context
2. When an atom is watched through this context
3. When watching an atom is initiated through this context

File: Sources/Atoms/Context/AtomTransactionContext.swift [hunk 2]
```diff
@@ -155,13 +155,13 @@ public struct AtomTransactionContext<Coordinator>: AtomWatchableContext {
         _store.reset(atom)
     }
 
-    /// Accesses the value associated with the given atom for reading and initialing watch to
+    /// Accesses the value associated with the given atom for reading and initializes watch to
```
Comment (rasberik) [inline Sources/Atoms/Context/AtomTransactionContext.swift:158]: [Q] initiates?

Comment (rasberik) [reply Sources/Atoms/Context/AtomTransactionContext.swift]: Seems so, updated

File: Sources/Atoms/PropertyWrapper/ViewContext.swift [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import Combine
 import SwiftUI
 
-/// A property wrapper type that provides a context structure that to read, watch, and otherwise
-/// interacting with atoms from views.
+/// A property wrapper type that provides a context structure to read, watch, and otherwise
+/// interact with atoms from views.
 ///
-/// Through the provided context, the view can read, write, or some other interactions to atoms.
+/// Through the provided context, the view can read, write, or perform other interactions with atoms.
```
Comment (rasberik) [inline Sources/Atoms/PropertyWrapper/ViewContext.swift:7]: Was missing a verb

File: Sources/Atoms/PropertyWrapper/Watch.swift [hunk 4]
```diff
@@ -28,7 +28,7 @@ public struct Watch<Node: Atom>: DynamicProperty {
     @ViewContext
     private var context
 
-    /// Creates a watch with the atom that to be watched.
+    /// Creates a watch with the atom to be watched.
```
Comment (rasberik) [inline Sources/Atoms/PropertyWrapper/Watch.swift:31]: > Creates a watch with the atom to be watched.

May be better to revisit this.
[Options]
1. Initializer with the atom to be watched.
2. Creates a watch property wrapper with the atom to be watched.

File: Sources/Atoms/Context/AtomTransactionContext.swift (hunk 1 above)
Comment (ra1028) [reply Sources/Atoms/Context/AtomTransactionContext.swift:3]: I'd vote for 2!

Comment (ra1028) [reply Sources/Atoms/Context/AtomTransactionContext.swift:3]: I'd vote for 2!

File: Sources/Atoms/PropertyWrapper/Watch.swift (hunk 4 above)
Comment (ra1028) [reply Sources/Atoms/PropertyWrapper/Watch.swift:31]: How about `Creates an instance with the atom to be watched.`?

**Review by ra1028 — COMMENTED** [review]
As for the potential improvement points;

> "Parameter atom: An atom that associates the value." (repetitive) → find it confusing, since atoms do associate with value, but not always other way around.

Agreed.
It probably better to have dedicated description per methods while keeping it a very brief description, e.g.
- `watch`: An atom to be watched.
- `read`: An atom to read the value

> "notifies update to downstream atoms" (repetitive) → can be better without update, or move it to back of the sentence

Agreed to remove update.

> Remove pronouns, e.g. "you" → In documentation, might be better avoid using pronouns,

I had the same idea but I vaguely remember that Apple's doc also contains pronouns such as "you" and that's why I used it.
However, I actually don't want to use it in documentation in general so feel free to replace them.

**Review by ra1028 — APPROVED** [review]
Beautiful! 💯 Thank you @rasberik 
The changes will automatically be reflected to our documentation page when next tag release.

**Pull request merged**